plugins/inputs/procstat/procstat.go

```diff
@@ -287,13 +287,21 @@ func (p *Procstat) updateProcesses(pids []PID, tags map[string]string, prevInfo
 	for _, pid := range pids {
 		info, ok := prevInfo[pid]
 		if ok {
+			// Assumption: if a process has no name, it probably does not exist
+			if name, _ := info.Name(); name == "" {
```

Comment:
Can we use `err != nil` here instead?  I feel like this would be safer than assuming a process cannot have a name.

Reply:
I've changed the lines below to:
```golang
// Assumption: if a process has no name, it probably does not exist
name, err := proc.Name()
if err != nil {
	fmt.Println("proc.Name() gave error: ", err)
}
if name == "" {
	fmt.Println("proc.Name() was empty, but no error!")
	continue
}
procs[pid] = proc
```
It results in the following output:
```
┌─[Kita][⚡ 93%][10:58:49] in ~/Projects/go/src/github.com/influxdata/telegraf
└─⎇ verify-pidfile-process-exists U:1 ?:1 ✗ ⋗ ./telegraf --config telegraf.conf --test
2019-04-26T08:58:59Z I! Starting Telegraf
proc.Name() was empty, but no error!
proc.Name() was empty, but no error!
> procstat_lookup,host=Kita.local,pid_finder=pgrep,pidfile=/tmp/process.pid,result=success pid_count=1i,result_code=0i,running=0i 1556269140000000000
```
So we do not receive an error, even when the process does not exist.

Reply:
Which operating system is that with?

Reply:
This is on Darwin.

Reply:
I wonder if this could be a bug in the Darwin implementation of gopsutil.  I tried this on FreeBSD, which has a very similar implementation, and I did get a "no such process" `syscall.Errno` error.  Can you verify that the sysctl syscall is returning no error?

https://github.com/shirou/gopsutil/blob/master/process/process_darwin.go#L627

For comparision, FreeBSD call is here:

https://github.com/shirou/gopsutil/blob/master/internal/common/common_freebsd.go#L54

Reply:
I get the following output when I add `fmt.Println("Syserr is: ", syserr)` to `process_darwin.go`:
```
Syserr is:  errno 0
proc.Name() was empty, but no error!
```
So it does not throw an error...

Reply:
Lets go with what we have here then.